pkg/apis/security/v1beta1/types.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterNetworkPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of ClusterNetworkPolicy.
+	Spec ClusterNetworkPolicySpec `json:"spec"`
+}
+
+// ClusterNetworkPolicySpec defines the desired state for ClusterNetworkPolicy.
+type ClusterNetworkPolicySpec struct {
+	// Priority specfies the order of the ClusterNetworkPolicy relative to
+	// other ClusterNetworkPolicies.
+	Priority int32 `json:"priority"`
```

Comment:
In the ofctrl module that we use, priority of a flow is defined by unit16
type FlowMatch struct {
	Priority     uint16            // Priority of the flow
since there are at most 65k unique priorities within a table. Theoretically we cannot have more rule priorities than OF priorities in the table, so maybe uint16 is sufficient here?

Reply:
I remember @McCodeman suggesting that we use a float value here so that:
* a NetworkPolicy with a priority P3 can always be created such that P1 < P3 < P2, as long as P1 < P2; this means that users never need to change existing priority values when they need to insert a policy in-between 2 existing policies with contiguous priority values (e.g. 3 & 4).
* UIs that let you "drag-and-drop" policies can easily create these CNP CRDs by mapping position on-screen to a float priority value

@abhiraut what do you think?

Reply:
thanks for reminding Antonin. I think we need to keep the user facing API with float64 (to accommodate for auto generated priorities when a UI is available) and internally in agent manage the conversion to its appropriate limits

Reply:
updated to float64